x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java

```diff
@@ -313,6 +352,18 @@ public synchronized void triggered(Event event) {
         }
 
         logger.debug("Data frame indexer [{}] schedule has triggered, state: [{}]", event.getJobName(), getIndexer().getState());
+        // If we are failed, don't trigger
+        if (taskState.get() == DataFrameTransformTaskState.FAILED) {
```

Comment:
Having these checks earlier (they are done lower down in the indexer as well) allow for simpler reasoning around what is done within these `synchronized` methods of `stop`, `start`, and `triggered`